processor/src/decoder/mod.rs

```diff
@@ -249,6 +254,8 @@ impl Decoder {
             aux_hints: AuxTraceHints::new(),
             in_debug_mode,
             operations: Vec::<Operation>::new(),
+            asmop_list: Vec::<(usize, AsmOpInfo)>::new(),
+            asmop_idx: 0,
```

Comment:
It seems like `asmop_idx` is used to help with iteration. Why not put it into `VmStateIterator`? This would also make it possible not to use mutable reference to `self` ins `get_asmop_at()`.

Reply:
Thanks. Fixed.